app/components/viral/form/help_text_component.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Viral
+  module Form
+    # Component to render form help text can be either default, success or error
+    class HelpTextComponent < ViewComponent::Base
+      attr_reader :icon
+
+      STATE_MAPPINGS = {
+        default: 'mt-2 text-sm text-slate-500 dark:text-slate-400',
+        success: 'mt-2 text-sm text-green-600 dark:text-green-500',
+        error: 'mt-2 text-sm text-red-600 dark:text-red-500'
+      }.freeze
+
+      ICON_MAPPINGS = {
+        default: 'information_circle_solid',
+        success: 'check_circle_solid',
+        error: 'x_circle'
+      }.freeze
+
+      def initialize(state: :default, **system_arguments)
+        @state = state
+        @arguments = system_arguments
+        @icon = ICON_MAPPINGS.key?(@state) ? ICON_MAPPINGS[@state] : ICON_MAPPINGS[:default]
+      end
+
+      def system_arguments
+        @arguments.tap do |args|
```

Comment:
This is a little embarrassing, but I've never heard about or seen the method tap before, lol.

Reply:
Yeah, it's pretty handy.  I think I first saw it in the Polaris code base.  Also a small right up on it here: https://medium.com/aviabird/ruby-tap-that-method-90c8a801fd6a